### DISCUSSION TITLE
OpenTelemetry DevServices

### DISCUSSION THREAD

**Discussion by radcortez** [quarkusio/quarkus#24373] [Community]
Is there any interest in adding DevServices to the OpenTelemetry extension? Thinking about the OTel Collector, plus an exporter like Jaeger (or something else).

//cc @knutwannheden @luneo7 @fabrii @XSAM

**Comment by knutwannheden** [comment]
Interesting question. I suppose the use case would be to allow inspecting and fine-tuning the trace and metrics sent by a service.

Another option could be to configure an in-memory exporter and then display the collected data in the Dev UI. Although the UI would have to be very primitive, because that would otherwise be a lot of work. But I think that could work, because the developer using this is likely going to be interested in the details. But this may still be too much work.

**Reply by radcortez** [reply]
> Interesting question. I suppose the use case would be to allow inspecting and fine-tuning the trace and metrics sent by a service.

Yes.

> Another option could be to configure an in-memory exporter and then display the collected data in the Dev UI. Although the UI would have to be very primitive, because that would otherwise be a lot of work. But I think that could work, because the developer using this is likely going to be interested in the details. But this may still be too much work.

Adding the in-memory exporter is easy (we do this for tests). The Dev UI idea is interesting. My concern here is that it may be too much work to properly display Tracers and Span (considering other tools that already do that properly).

**Comment by fabrii** [comment]
I'll give my opinion related to when I tried to use the RedPanda Kafka DevService.
I wanted to integrate Kafka with Kafka Connect and ElasticSearch. I had Kafka Connect and ElasticSearch defined in a docker-compose file. The biggest issue was the connectivity between the DevService Kafka with the other two components. They were not sharing the network. At the end, it was easier to just make a docker-compose with the 3 services and start them together with docker-compose up.
I wonder if something similar can happen in this case. I also think that there are some WIP issues related to OpenTelemetry that could be more useful.

**Reply by radcortez** [reply]
> I'll give my opinion related to when I tried to use the RedPanda Kafka DevService.
> I wanted to integrate Kafka with Kafka Connect and ElasticSearch. I had Kafka Connect and ElasticSearch defined in a docker-compose file. The biggest issue was the connectivity between the DevService Kafka with the other two components. They were not sharing the network. At the end, it was easier to just make a docker-compose with the 3 services and start them together with docker-compose up.

In this case, we would bootstrap everything required by OTel by DevServices.

> I also think that there are some WIP issues related to OpenTelemetry that could be more useful.

This is just a discussion. It doesn't mean it will get higher priority to other stuff already in the backlog.
